### PR TITLE
tool: convert str to num using std::stoull()

### PR DESCRIPTION
**Pull request by simon-rock**
Signed-off-by: Simon Gao <simon29rock@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/tools/cephfs/MetaTool.cc [hunk 1]
```diff
@@ -773,10 +773,7 @@ int MetaTool::list_meta(meta_op &op)
 }
 unsigned long long MetaTool::conv2hexino(const char* ino)
 {
-  unsigned long long iino = 0;
-  std::stringstream conv;
-  conv << ino;
-  conv >> iino;
+  unsigned long long iino = strtoull (ino, NULL, 0);
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:776]: i'd suggest just drop this helper and use `std::stoull()` instead.

Comment (simon-rock) [reply src/tools/cephfs/MetaTool.cc:776]: We will get the decimal format of ino, When problems happen.
but we need hexadecimal format of ino to find it from memory cache, mds dump ino, or from disk using this tool.
That's why I added this method.

Comment (simon-rock) [reply src/tools/cephfs/MetaTool.cc]: updated it.
Now this method is only for finding the file from disk.

File: src/tools/cephfs/MetaTool.cc [hunk 2]
```diff
@@ -773,10 +773,7 @@ int MetaTool::list_meta(meta_op &op)
 }
 unsigned long long MetaTool::conv2hexino(const char* ino)
 {
-  unsigned long long iino = 0;
-  std::stringstream conv;
-  conv << ino;
-  conv >> iino;
+  unsigned long long iino = std::stoull(ino, NULL, 0);
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:776]: could you drop `MetaTool::conv2hexino()` ? see:

```
git grep -w conv2hexino                                                                                                                                    
src/tools/cephfs/MetaTool.cc:    inodeno_t i_ino = conv2hexino(ino.c_str());
src/tools/cephfs/MetaTool.cc:    inodeno_t i_ino = conv2hexino(ino.c_str());
src/tools/cephfs/MetaTool.cc:    inodeno_t i_ino = conv2hexino(ino.c_str());
src/tools/cephfs/MetaTool.cc:    inodeno_t i_ino = conv2hexino(ino.c_str());
```

i don't see the point having it.

File: src/tools/cephfs/MetaTool.cc [hunk 3]
```diff
@@ -207,7 +207,7 @@ int MetaTool::process(string& mode, string& ino, string out, string in, bool con
 int MetaTool::show_fnode(string& ino, string& out)
 {
   if (ino != "0") {
-    inodeno_t i_ino = conv2hexino(ino.c_str());
+    inodeno_t i_ino = std::stoull(ino.c_str(), NULL, 0);
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:210]:
```suggestion
    inodeno_t i_ino = std::stoull(ino, NULL, 0);
```

Comment (simon-rock) [reply src/tools/cephfs/MetaTool.cc]: @tchaikov updated it

**Comment by batrick** [comment]
jenkins retest this please

**Pull request merged**